include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1167 @@
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
+            return false;
+        }
+
+        // Set Options for what events we get from Google
+        $optParams = array(
+            'maxResults' => 2, // We ask for a max of 2 because we should *NEVER* get more than 1. If we get 2, something is horribly wrong.
+            'showDeleted' => true,
+            'singleEvents' => false,
+            'timeMin' => date('c', strtotime('-6 month')),
+            'privateExtendedProperty' => 'suitecrm_id=' . $meeting_id
+        );
+
+        $results_g = $this->gService->events->listEvents($this->calendarId, $optParams);
+        $results = $results_g->getItems();
+
+        if (count($results) > 1) {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'We received more than one Google event with the same SuiteCRM meeting ID. Something is horribly wrong!');
+            return false;
+        } elseif (count($results) == 0) { // No events match. Return emtpy array.
+            return array();
+        } else {
+            $gEvent = $results[0];
+            return $gEvent;
+        }
+    }
+
+    /**
+     * Get a google event by the event id
+     *
+     * @param string $event_id Google Event ID
+     *
+     * @return \Google_Service_Calendar_Event|null Google_Service_Calendar_Event if found, null if not found
+     */
+    public function getGoogleEventById($event_id)
+    {
+
+        // Make sure the calendar service is set up
+        if (!isset($this->gService)) {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
+            return false;
+        }
+
+        if (empty($event_id)) {
+         // If we didn't get passed an event, return null
+            return;
+        }
+
+        $gEvent = $this->gService->events->get($this->calendarId, $event_id);
+
+        if (!empty($gEvent->getId())) {
+            return $gEvent;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Find a missing event on users calendars
+     *
+     * This is an expensive method. It searches *all* the users calendars for a missing event,
+     * moves it back to the SuiteCRM calendar, and returns the moved event.
+     *
+     * @param string $event_id The Google Event Id
+     *
+     * @return \Google_Service_Calendar_Event|null Google_Service_Calendar_Event if found, null if not found
+     */
+    public function getMissingMeeting($event_id)
+    {
+        // get list of users calendars
+        $calendarList = $this->gService->calendarList->listCalendarList();
+
+        // Search all calendars looking for event
+        foreach ($calendarList->getItems() as $calendarListEntry) {
+            if ($gEvent = $this->gService->events->get($calendarListEntry->getId(), $event_id)) {
```

Comment:
This entire method is currently unused. I'm not sure it will be used in the future. I created it because I was concerned about what would happen if a user moved a synced event to a different calendar. It turns out that the Google Event ID stays the same even if it's moved, so this was never needed. I kept it around in case I found another use for it in the future. The variable was set for testing, so I could see the Google Event ID in xdebug.
Not sure what, if anything, to do with this. It could be removed, but I may use it in the future when we get more complicated features.

Reply:
Ok, if you need this variable by any reason but in case please.
"Using the assignment operator in conditional expressions frequently indicates programmer error and can result in unexpected behavior."
Use the variable value assignment before the if condition.
Compilant solution:
```
  a = b;
  if (a) {
    /* ... */
  }
```

Reply:
I see what you mean. If I ever need it, I'll do that. For now I'm going to remove the method from production code, since it's not used anywhere, and is untested.